lib/services/android-plugin-build-service.ts

```diff
@@ -164,113 +169,157 @@ export class AndroidPluginBuildService implements IAndroidPluginBuildService {
 	 */
 	public async buildAar(options: IBuildOptions): Promise<boolean> {
 		this.validateOptions(options);
+		const manifestFilePath = this.getManifest(options.platformsAndroidDirPath);
+		const androidSourceSetDirectories = this.getAndroidSourceDirectories(options.platformsAndroidDirPath);
+		const shouldBuildAar = !!manifestFilePath || androidSourceSetDirectories.length > 0;
+
+		if (shouldBuildAar) {
+			const shortPluginName = getShortPluginName(options.pluginName);
+			const pluginTempDir = path.join(options.tempPluginDirPath, shortPluginName);
+			const pluginTempMainSrcDir = path.join(pluginTempDir, "src", "main");
+
+			await this.updateManifest(manifestFilePath, pluginTempMainSrcDir, shortPluginName);
+			this.copySourceSetDirectories(androidSourceSetDirectories, pluginTempMainSrcDir);
+			await this.setupGradle(pluginTempDir, options.platformsAndroidDirPath, options.projectDir);
+			await this.buildPlugin({ pluginDir: pluginTempDir, pluginName: options.pluginName });
+			this.copyAar(shortPluginName, pluginTempDir, options.aarOutputDir);
+		}
 
-		// IDEA: Accept as an additional parameter the Android Support Library version from CLI, pass it on as -PsupportVersion later to the build
-		// IDEA: apply app.gradle here in order to get any and all user-defined variables
-		// IDEA: apply the entire include.gradle here instead of copying over the repositories {} and dependencies {} scopes
+		return shouldBuildAar;
+	}
 
-		const shortPluginName = getShortPluginName(options.pluginName);
-		const newPluginDir = path.join(options.tempPluginDirPath, shortPluginName);
-		const newPluginMainSrcDir = path.join(newPluginDir, "src", "main");
+	private async updateManifest(manifestFilePath: string, pluginTempMainSrcDir: string, shortPluginName: string): Promise<void> {
+		let updatedManifestContent;
+		this.$fs.ensureDirectoryExists(pluginTempMainSrcDir);
 		const defaultPackageName = "org.nativescript." + shortPluginName;
+		if (manifestFilePath) {
+			let androidManifestContent;
+			try {
+				androidManifestContent = this.$fs.readText(manifestFilePath);
+			} catch (err) {
+				throw new Error(
+					`Failed to fs.readFileSync the manifest file located at ${manifestFilePath}`
```

Comment:
Use `this.$errors.fail` or `this.$erros.failWithoutHelp` instead `throw new Error`